apps/gitness/src/pages/pull-request/context/pull-request-data-provider.tsx

```diff
@@ -109,7 +109,7 @@ const PullRequestDataProvider: React.FC<PullRequestDataProviderProps> = ({ child
   const repoRef = useGetRepoRef()
   const { data: repoMetadata } = useFindRepositoryQuery({ repo_ref: repoRef })
   const { pullRequestId } = useParams<PathParams>()
-
+  const pullRequestSection = window.location.href.split(pullRequestId + '/')[1]
```

Comment:
Can we do route match here instead? Something like:

https://github.com/harness/canary/blob/signin-page-redirect-repeatedly-issue/apps/gitness/src/layouts/PullRequestLayout.tsx#L32

Reply:
I'll refactor this into a hook later on.

Reply:
Should I leave this as is or remove it  @vardanbansal-harness

Reply:
Let's add route match here for now, location parsing looks hacky.

Reply:
Added it as part of this PR https://github.com/harness/canary/pull/287. Pls take a look.